source/custom/rules/hostname.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package rules
+
+import (
+	"log"
+	"os"
+	"regexp"
+)
+
+var (
+	// TODO is it safe to assume nodeName == hostname??
+	nodeName = os.Getenv("NODE_NAME")
```

Comment:
any reason not to use `/etc/hostname` ?

`NODE_NAME` is set as the k8s node object name, im sure there is some kubelet flag that allows to advertise the node under a different name somewhere.

from kubelet --help:
`--hostname-override string                                 If non-empty, will use this string as identification instead of the actual hostname.`

If these are indeed non synonymous maybe the Rule should be NodeNameRule ?

Reply:
Very good point, thanks! I need to clarify this.

Reply:
after clarifying I switch to nodename everywhere :)